File: code/video/MyCloudMusicAndroidJava/app/src/main/java/com/ixuea/courses/mymusic/component/user/activity/UserDetailActivity.java
package com.ixuea.courses.mymusic.component.user.activity;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class UserDetailActivity extends AppCompatActivity {
    public static void startWithId(Context context, String id) {
        Intent intent = new Intent(context, UserDetailActivity.class);
        intent.putExtra("id", id);
        context.startActivity(intent);
    }

    public static void startWithNickname(Context context, String nickname) {
        Intent intent = new Intent(context, UserDetailActivity.class);
        intent.putExtra("nickname", nickname);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        finish();
    }
}

File: code/video/MyCloudMusicAndroidJava/app/src/main/java/com/ixuea/courses/mymusic/component/user/activity/UserDetailActivity.kt
package com.ixuea.courses.mymusic.component.user.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class UserDetailActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        finish()
    }

    companion object {
        @JvmStatic
        fun startWithId(context: Context, id: String?) {
            val intent = Intent(context, UserDetailActivity::class.java).apply {
                putExtra("id", id)
            }
            context.startActivity(intent)
        }

        @JvmStatic
        fun startWithNickname(context: Context, nickname: String?) {
            val intent = Intent(context, UserDetailActivity::class.java).apply {
                putExtra("nickname", nickname)
            }
            context.startActivity(intent)
        }
    }
}
